src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -582,16 +582,31 @@ IEnumerable<FunctionSymbol> GetAccessibleDecoratorFunctionsWithCache(NamespaceTy
 
             if (!context.Kind.HasFlag(BicepCompletionContextKind.DecoratorName))
             {
-                // add namespaces first
-                AddSymbolCompletions(completions, nsTypeDict.Keys);
+                if (!context.Kind.HasFlag(BicepCompletionContextKind.SymbolicName))
+                {
+                    // add namespaces first
+                    AddSymbolCompletions(completions, nsTypeDict.Keys);
+                }
 
                 // add accessible symbols from innermost scope and then move to outer scopes
                 // reverse loop iteration
                 for (int depth = context.ActiveScopes.Length - 1; depth >= 0; depth--)
```

Comment:
@majastrz does the depth matter for the cases we care about? (Refer to the test added for use cases)